src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -553,6 +564,20 @@ def _build_pex_description(request: PexRequest) -> str:
     return f"Building {request.output_filename} {desc_suffix}"
 
 
+def _pex_lockfile_requirement_count(lock_content: str) -> int:
+    # TODO: this is a very naive heuristic that will overcount, and also relies on Pants
+    #  setting `--indent` when generating lockfiles. More robust would be parsing the JSON
+    #  and getting the len(locked_resolves.locked_requirements.project_name), but we risk
+    #  if Pex ever changes its lockfile format.
+
+    num_lines = len(lock_content.splitlines())
+    # These are very naive estimates, and they bias towards overcounting. For example, requirements
+    # often are 20+ lines.
+    num_lines_for_options = 10
+    lines_per_req = 10
+    return max((num_lines - num_lines_for_options) // lines_per_req, 2)
```

Comment:
How do people feel about counting the # of `project_name`s there are? :thinking: